File: component/src/main/java/com/mackwu/component/JetpackFragment.java
package com.mackwu.component;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.mackwu.component.jetpack.UserLiveData;
import com.mackwu.xmvc.BaseFragment;

/**
 * ===================================================
 * Created by MackWu on 2020/6/19 23:43
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 */
public class JetpackFragment extends BaseFragment {

    private Button btnLifeCycle;
    private UserLiveData userLiveData;

    @Override
    public int getLayoutId() {
        return R.layout.fragment_jetpack;
    }

    @Override
    public void initView(@NonNull View view, @Nullable Bundle savedInstanceState) {

    }

    @Override
    public void initData(@Nullable Bundle savedInstanceState) {

    }

}

File: component/src/main/java/com/mackwu/component/JetpackFragment.kt
package com.mackwu.component

import com.mackwu.component.fragment.base.SimpleBaseFragment
import com.mackwu.component.jetpack.viewmodel.TestViewModelActivity
import com.mackwu.component.util.startActivityCls
import kotlinx.android.synthetic.main.fragment_jetpack.*

/**
 * ===================================================
 * Created by MackWu on 2020/4/17 10:31
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 */
class JetpackFragment : SimpleBaseFragment() {

    override val layoutId: Int = R.layout.fragment_jetpack

    override fun initView() {
        btn_view_model.setOnClickListener { startActivityCls(TestViewModelActivity::class.java) }
    }
}